packages/composer/lib/hooks.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+
+async function composeOpenAPI (app, opts) {
+  const onSendHooks = {}
+
+  app.addHook('onSend', async (req, reply, payload) => {
+    const method = req.routeConfig.method
+    const openapiPath = req.routeConfig.openapiPath
+
+    const hooks = onSendHooks[openapiPath]?.[method]
```

Comment:
This is better set on only the matching routes, instead of running for all routes.

Reply:
The problem is that we set user plugins after composing routes. And this is the correct order.